### PR TITLE
BABEL-745 Support STDEV(), STDEVP(), VAR(), VARP() statistical T-SQL

### PR DESCRIPTION
**Pull request by pratikzode**
STDEV()
STDEV function is used in SQL to calculate the standard deviation of a set of values in a given column of a table. STDEV stands for "standard deviation," which is a statistical measure that represents the degree of variation of a set of values from its mean or average

STDEVP()
STDEVP function is similar to the STDEV function in SQL, but it calculates the standard deviation of an entire population rather than a sample of the population. "STDEVP" stands for "standard deviation of a population".

VAR()
VAR function is used in SQL to calculate the variance of a set of values in a given column of a table. "VAR" stands for "variance," which is a statistical measure that represents how much a set of values varies from its mean or average.

VARP()
VARP function is used in SQL to calculate the variance of an entire population rather than a sample of the population. "VARP" stands for "variance of a population"

Task: BABEL-745

### CODE REVIEW

File: test/JDBC/expected/BABEL-745-vu-prepare.out [hunk 1]
```diff
@@ -0,0 +1,67 @@
+-- Create SalesData table
+CREATE TABLE SalesData (
+  Product varchar(50),
+  SalesAmount decimal(10,2)
```
Comment (macwoody) [inline test/JDBC/expected/BABEL-745-vu-prepare.out]: your function signature is receiving float value. 
So please also test with float value

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.1.0--3.2.0.sql [hunk 2]
```diff
@@ -78,3 +78,38 @@ DROP FUNCTION sys.pg_extension_config_remove(REGCLASS);
 -- Reset search_path to not affect any subsequent scripts
 SELECT set_config('search_path', trim(leading 'sys, ' from current_setting('search_path')), false);
 
+CREATE AGGREGATE STDEV(float8) (
```
Comment (macwoody) [inline contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.1.0--3.2.0.sql]: please specify sys schema for your function definitions

File: test/JDBC/input/BABEL-745-vu-prepare.sql [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+-- Create a stored procedure for VAR calculation
+CREATE PROCEDURE sp2_CalculateStdev
+  @Product varchar(50)
+AS
+BEGIN
+  SELECT VAR(SalesAmount) AS SalesAmountStdev
+  FROM SalesData
+  WHERE Product = @Product
+  GROUP BY Product;
```
Comment (macwoody) [inline test/JDBC/input/BABEL-745-vu-prepare.sql:74]: for all select statement you are using, please add order by clause

File: test/JDBC/input/BABEL-745-vu-verify.sql [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+GO
+
+SELECT Product, VAR(CAST(SalesAmount AS float8)) AS SalesAmountStdev
+FROM SalesData
+GROUP BY Product
+GO
+
+SELECT Product, VARP(CAST(SalesAmount AS float8)) AS SalesAmountStdev
+FROM SalesData
+GROUP BY Product
```
Comment (macwoody) [inline test/JDBC/input/BABEL-745-vu-verify.sql:34]: ORDER BY

File: test/JDBC/input/BABEL-745-vu-prepare.sql [hunk 5]
```diff
@@ -0,0 +1,65 @@
+-- Create SalesData table
+CREATE TABLE SalesData (
+  Product varchar(50),
+  SalesAmount decimal(10,2)
+);
+-- Insert some sample data
+INSERT INTO SalesData (Product, SalesAmount)
```
Comment (macwoody) [inline test/JDBC/input/BABEL-745-vu-prepare.sql:7]: please bring PG's existing tests here in addition to the scenario you created

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.1.0--3.2.0.sql (hunk 2 above)
Comment (xhfanhe) [reply contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.1.0--3.2.0.sql]: And please put everything before line 73
"-- Drops the temporary procedure used by the upgrade script."

File: test/JDBC/upgrade/latest/schedule [hunk 6]
```diff
@@ -407,6 +407,7 @@ openquery_upgrd
 BABEL-3657
 test_windows_alter_user
 BABEL-733
+BABEL-745
```
Comment (surendravishnoi) [inline test/JDBC/upgrade/latest/schedule:410]: adding to schedule will not exercise 3.1 to 3.2, add it to 14.6 or 14.7 as well

File: test/JDBC/expected/BABEL-745-vu-prepare.out [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+INSERT INTO SalesData (Product, SalesAmount)
+VALUES ('Product A', 100.00), ('Product A', 150.00), ('Product A', 200.00),
+       ('Product B', 50.00), ('Product B', 75.00), ('Product B', 100.00),
+       ('Product C', 25.00), ('Product C', 50.00), ('Product C', 75.00);
+GO
+~~ROW COUNT: 9~~
+
+
+-- Create a view for SalesData
+CREATE VIEW v_SalesData AS
```
Comment (macwoody) [inline test/JDBC/expected/BABEL-745-vu-prepare.out:16]: The view definition has to contain the function definition, just like below store procedures

File: test/JDBC/expected/BABEL-745-vu-verify.out [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+
+EXEC sp3_CalculateStdev 'Product A';
+GO
+~~START~~
+float
+1666.6666666666667
+~~END~~
+
+
+SELECT
```
Comment (macwoody) [inline test/JDBC/expected/BABEL-745-vu-verify.out]: What is intention of this test ?

File: test/JDBC/expected/BABEL-745-vu-verify.out [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+
+
+SELECT VARP(SalesAmount) AS SalesAmountStdev
+FROM SalesData
+GROUP BY Product
+ORDER BY Product ASC;
+GO
+~~START~~
+float
+1666.6666666666667
```
Comment (lejaokri) [inline test/JDBC/expected/BABEL-745-vu-verify.out:98]: quick question: these output were verified against sql server, right?

Comment (pratikzode) [reply test/JDBC/expected/BABEL-745-vu-verify.out:98]: Yes

File: test/JDBC/expected/BABEL-745-vu-prepare.out [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+SELECT Product, STDEV(SalesAmount) AS SalesAmount
```
Comment (macwoody) [inline test/JDBC/expected/BABEL-745-vu-prepare.out:17]: why just stdev ? plz have all other functions covered by view as well.

Comment (pratikzode) [reply test/JDBC/expected/BABEL-745-vu-prepare.out:17]: Yes Covered it in new commit

**Review by macwoody — APPROVED** [review]

**Review by xhfanhe — APPROVED** [review]

**Pull request merged**